packages/contracts/scripts/minter-deployments/shared-minters-deployer.ts

```diff
@@ -0,0 +1,195 @@
+      );
+    }
+    // check minter filter address is defined
+    if (!ethers.utils.isAddress(deployDetails.minterFilterAddress)) {
+      throw new Error(
+        `[ERROR] minterFilterAddress must be defined at index ${index}, but is ${deployDetails.minterFilterAddress}`
+      );
+    }
+  }
+  //////////////////////////////////////////////////////////////////////////////
+  // INPUT VERIFICATION ENDS HERE
+  //////////////////////////////////////////////////////////////////////////////
+
+  // Perform the following steps for each to-be-deployed contract
+  for (let index = 0; index < deployConfigDetailsArray.length; index++) {
+    const deployDetails = deployConfigDetailsArray[index];
+
+    //////////////////////////////////////////////////////////////////////////////
+    // DEPLOYMENT BEGINS HERE
+    //////////////////////////////////////////////////////////////////////////////
+
+    // deploy new shared minter contract
+    const minterName = deployDetails.minterName;
+    const minterConstructorArgs = [deployDetails.minterFilterAddress];
+    // add delegation registry address to constructor args if needed
+    if (
+      minterName.includes("Holder") ||
+      minterName.includes("Merkle") ||
+      minterName.includes("Polyptych")
+    ) {
+      const delegationRegistryAddress =
+        DELEGATION_REGISTRY_ADDRESSES[networkName];
+      if (!delegationRegistryAddress) {
+        throw new Error(
+          `[ERROR] delegationRegistryAddress must be defined on network ${networkName}, but is ${delegationRegistryAddress}`
+        );
+      }
+      minterConstructorArgs.push(DELEGATION_REGISTRY_ADDRESSES[networkName]);
+    }
+    const minterFactory = await ethers.getContractFactory(minterName);
+    const minter = await minterFactory.deploy(...minterConstructorArgs);
+    await minter.deployed();
+    const minterAddress = minter.address;
+    console.log(
+      `[INFO] ${deployDetails.minterName} deployed at ${minterAddress}`
+    );
+    await delay(EXTRA_DELAY_BETWEEN_TX);
+
+    //////////////////////////////////////////////////////////////////////////////
+    // DEPLOYMENT ENDS HERE
+    //////////////////////////////////////////////////////////////////////////////
+
+    //////////////////////////////////////////////////////////////////////////////
+    // SETUP BEGINS HERE
+    //////////////////////////////////////////////////////////////////////////////
+
+    if (deployDetails.approveMinterGlobally) {
+      // approve minter globally
+      const minterFilter = await ethers.getContractAt(
+        ASSUMED_MINTER_FILTER_NAME,
+        deployDetails.minterFilterAddress
+      );
+      const tx = await minterFilter.approveMinterGlobally(minterAddress);
+      await tx.wait();
+      console.log(
+        `[INFO] ${deployDetails.minterName} approved globally on minter filter at ${deployDetails.minterFilterAddress}`
+      );
+      await delay(EXTRA_DELAY_BETWEEN_TX);
+    }
```

Comment:
I think we should try catch here – to gracefully fail when the connected wallet doesn't actually have the ability to perform this step.

Reply:
good call - updated in [d077ad5](https://github.com/ArtBlocks/artblocks-contracts/pull/901/commits/d077ad50496abcbac2326631ccd3f60b969bec75)